source/mqtt_request_response.c

```diff
@@ -481,11 +483,198 @@ PyObject *aws_py_mqtt_request_response_client_make_request(PyObject *self, PyObj
     return result;
 }
 
+/***************************************************************************************/
+
+struct mqtt_streaming_operation_binding {
+    struct aws_mqtt_rr_client_operation *native;
+    PyObject *subscription_status_changed_callable;
+    PyObject *incoming_publish_callable;
+};
+
+static struct mqtt_streaming_operation_binding *s_mqtt_streaming_operation_binding_new(
+    PyObject *subscription_status_changed_callable_py,
+    PyObject *incoming_publish_callable_py) {
+    struct mqtt_streaming_operation_binding *binding =
+        aws_mem_calloc(aws_py_get_allocator(), 1, sizeof(struct mqtt_streaming_operation_binding));
+
+    binding->subscription_status_changed_callable = subscription_status_changed_callable_py;
+    Py_XINCREF(binding->subscription_status_changed_callable);
+
+    binding->incoming_publish_callable = incoming_publish_callable_py;
+    Py_XINCREF(binding->incoming_publish_callable);
+
+    return binding;
+}
+
+static void s_mqtt_streaming_operation_binding_on_terminated(void *user_data) {
+    struct mqtt_streaming_operation_binding *stream_binding = user_data;
+
+    PyGILState_STATE state;
+    if (!aws_py_gilstate_ensure(&state)) {
+        Py_XDECREF(stream_binding->subscription_status_changed_callable);
+        Py_XDECREF(stream_binding->incoming_publish_callable);
+
+        PyGILState_Release(state);
+    }
+
+    aws_mem_release(aws_py_get_allocator(), stream_binding);
+}
+
+static void s_mqtt_streaming_operation_binding_destructor(PyObject *stream_capsule) {
+    struct mqtt_streaming_operation_binding *stream_binding =
+        PyCapsule_GetPointer(stream_capsule, s_capsule_name_mqtt_streaming_operation);
+    assert(stream_binding);
+
+    stream_binding->native = aws_mqtt_rr_client_operation_release(stream_binding->native);
+}
+
+static void s_aws_mqtt_streaming_operation_subscription_status_callback_python(
+    enum aws_rr_streaming_subscription_event_type status,
+    int error_code,
+    void *user_data) {
+
+    struct mqtt_streaming_operation_binding *stream_binding = user_data;
+
+    PyGILState_STATE state;
+    if (aws_py_gilstate_ensure(&state)) {
+        return;
+    }
+
+    PyObject *result = PyObject_CallFunction(
```

Comment:
This seems to be a behavior difference between python and other SDK. For CPP SDK, if `aws_mqtt_rr_client_operation_release` is called, no callback would be invoked anymore. 
https://github.com/awslabs/aws-crt-cpp/blob/d792acb017f5596eb4356adff436aa18adaa727f/source/iot/MqttRequestResponseClient.cpp#L182

However, for python binding, there is no validation to stream_binding->native. Therefore, as long as the termination callback is not called, the callbacks will still be invoked regardless the python operation object is already destroyed.

Reply:
That's intentional; it's safe/fine for it to do so.

Reply:
My concern is more about the behavioral difference between SDKs rather than safety. In the CPP SDK, the callback is no longer invoked once the client is shut down, while in python, callbacks may still be triggered even after the client is destroyed.

I’m fine with either behavior, but it would be better if we keep them consistent across SDKs.